presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -287,8 +286,7 @@ public StageInfo getStageInfo()
         return stateMachine.getStageInfo(
                 () -> getAllTasks().stream()
                         .map(RemoteTask::getTaskInfo)
-                        .collect(toImmutableList()),
-                ImmutableList::of);
+                        .collect(toImmutableList()));
```

Comment:
I found `getBasicStageStats`/`getStageInfo` a bit hard to read, so how about we update them as:

```
public BasicStageStats getBasicStageStats()
{
    return stateMachine.getBasicStageStats(this::getAllTaskInfo);
}

public StageInfo getStageInfo()
{
    return stateMachine.getStageInfo(this::getAllTaskInfo);
}

private Iterable<TaskInfo> getAllTaskInfo()
{
    return getAllTasks().stream()
            .map(RemoteTask::getTaskInfo)
            .collect(toImmutableList());
}
```